pkg/controller/kfservice/kfservice_controller_test.go

```diff
@@ -39,35 +37,97 @@ import (
 
 var c client.Client
 
+const timeout = time.Second * 5
+
 var expectedRequest = reconcile.Request{NamespacedName: types.NamespacedName{Name: "foo", Namespace: "default"}}
 var serviceKey = types.NamespacedName{Name: "foo", Namespace: "default"}
-var kfserviceKey = types.NamespacedName{Name: "foo", Namespace: "default"}
 
-const timeout = time.Second * 5
+var expectedCanaryRequest = reconcile.Request{NamespacedName: types.NamespacedName{Name: "foo-canary", Namespace: "default"}}
+var canaryKey = types.NamespacedName{Name: "foo-canary", Namespace: "default"}
 
-func TestReconcile(t *testing.T) {
-	g := gomega.NewGomegaWithT(t)
-	instance := &servingv1alpha1.KFService{
-		ObjectMeta: metav1.ObjectMeta{
-			Name:      "foo",
-			Namespace: "default",
+var instance = &servingv1alpha1.KFService{
```

Comment:
There's a SampleTFProto in the types file. Do you think it would be a good pattern to import it for tests or better to be explicit?

Reply:
I prefer to be explicit in place as when the imported type changes test can get broken.